src/main/java/com/zoo/animals/Animal.java

```diff
@@ -30,6 +25,62 @@ public void setEnergyLevel(int energyLevel) {
 
     }
 
-    public abstract void makeSound();
+    private void setHappinessLevel(int happinessLevel) {
+        if (happinessLevel < 0) {
+            this.happinessLevel = 0;
+        } else if (happinessLevel > 100) {
+            this.happinessLevel = 100;
+        } else {
+            this.happinessLevel = happinessLevel;
+        }
+    }
+
+
+    protected void increaseEnergyLevelBy(int points) {
+        setEnergyLevel(this.energyLevel + points);
+    }
+
+    protected void decreaseEnergyLevelBy(int points) {
+        setEnergyLevel(this.energyLevel - points);
+    }
 
+    public void eat() {
+        increaseEnergyLevelBy(20);
+        setHappinessLevel(getHappinessLevel() + 5);
```

Comment:
not a mistake, although you see it also looks a bit awkward) - so maybe approach with adding methods like `increase/decreaseHappinessLevel` also could be used

But next time